src/orders/components/OrderFulfillPage/OrderFulfillPage.tsx

```diff
@@ -87,22 +96,37 @@ const OrderFulfillPage: React.FC<OrderFulfillPageProps> = props => {
 
   const { change: formsetChange, data: formsetData } = useFormset<
     null,
-    OrderFulfillStockInput[]
+    OrderFulfillLineFormData[]
   >(
     (getToFulfillOrderLines(order?.lines) as OrderFulfillLineFragment[]).map(
-      line => ({
-        data: null,
-        id: line.id,
-        label: getAttributesCaption(line?.variant?.attributes),
-        value: line?.variant?.preorder
-          ? null
-          : [
-              {
-                quantity: line.quantityToFulfill,
-                warehouse: warehouse?.id,
-              },
-            ],
-      }),
+      line => {
+        const highestQuantityAllocation = line.allocations.reduce(
+          (prevAllocation, allocation) => {
+            if (
+              !prevAllocation ||
+              prevAllocation.quantity < allocation.quantity
+            ) {
+              return allocation;
+            }
+            return prevAllocation;
+          },
+          null,
+        );
```

Comment:
I'd specify type of `highestQuantityAllocation` explicitly and move this to utils file